src/hw java8/src/main/java/com/company/hw/User.java

```diff
@@ -0,0 +1,183 @@
+package com.company.hw;
+
+import com.company.hw.books.Comics;
+import com.company.hw.books.Encyclopedia;
+import com.company.hw.books.Novel;
+import com.company.hw.exceptions.NoSuchBookException;
+import com.company.hw.sort.BookComparator;
+import com.company.hw.sort.YearComparator;
+import com.github.javafaker.Faker;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class User {
+
+    private List<Library> books;
+    LocalDate date = LocalDate.now();
+
+    Novel novelExample = new Novel("Example", "Example", 2019, 3, false);
+    Comics comicsExample = new Comics("Example", "Example", 2019, true);
+    Encyclopedia encyclopediaExample = new Encyclopedia("Example", "Example", 2012, true);
+
+
+
+    User() {
+        books = new ArrayList<>();
+        books.add(new Novel("N. Pidmogylnuy", "City", 1920, 3, false));
+        books.add(new Novel("M. Zerov", "Incognito", 1918, 3, false));
+        books.add(new Novel("B. Voronnuy", "Infanta", 1904, 3, true));
+        books.add(new Novel("E. Bronte", "Glass Town", 1827, 5, true));
+        books.add(new Novel("E. Bronte", "Wuthering Heights", 1847, 5, true));
+        books.add(new Novel("W. Shakespeare", "Gamlet", 1603, 1, false));
+        books.add(new Novel("T. Shevchenko", "B", 1865, 5, true));
+        books.add(new Novel("T. Shevchenko", "A", 1845, 5, true));
+        books.add(new Novel("T. Shevchenko", "C", 2019, 5, true));
+
+        books.add(new Comics("A. Chornuy", "Fly", 2019, true));
+        books.add(new Comics("K. Dankovych", "My life", 2017, false));
+        books.add(new Comics("H. Nazarov", "Green door of Earth", 2018, false));
+        books.add(new Comics("L. Poznyak", "Family", 1992, true));
+
+        books.add(new Encyclopedia("O. Bilyk", "Anatomy", 2012, true));
+        books.add(new Encyclopedia("P. Ivanov", "Ukrainian World", 2010, false));
+        books.add(new Encyclopedia("M. Orlyk", "The World Of Us", 2017, false));
+        books.add(new Encyclopedia("A. Kvadrat", "Geometry", 2001, true));
+        books.add(new Encyclopedia("Y. Bachynskyi", "Irredenta", 1920, false));
+        books.add(new Encyclopedia("A. Figure", "Universe of Algebra", 2011, true));
+
+    }
+
+    //1
+    void getAvailableBooks() {
+        LocalDate lastDay =
+                date.with(TemporalAdjusters.lastDayOfMonth());
+        DateTimeFormatter format =
+                DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        String formatedDateTime = lastDay.format(format);
+
+        books.stream().forEach(element -> {
```

Comment:
can we do stream->filter->foreach ... just to avoid IF

Reply:
FIXED